scripts/upgrade

```diff
@@ -1,128 +1,249 @@
 # See https://www.dokuwiki.org/install:unused_files
-if [ -f "../sources/data/deleted.files" ]; then
-	grep -Ev '^($|#)' ../sources/data/deleted.files | xargs -I {} sudo rm -vrf $final_path/{}
-fi
+if [ -f "$final_path/data/deleted.files" ]; then
 
-# Change owner for all plugins
-sudo chmod -R 755 $final_path/lib/plugins
+	# Use a "sub process" to start a new shell to run these commands
+	# Allow to use only one "cd" and to be more efficent
+	(
+		# Move to the dokuwiki installation folder so the "official" commands can be used without adaptation
+		cd $final_path
+
+		# This command could not remove directory
+		#grep -Ev '^($|#)' data/deleted.files | xargs -n 1 rm -vf
+		# => "rm: cannot remove 'vendor/easybook/geshi': Is a directory"
+
+		# That one works as expected
+		grep -Ev '^($|#)' data/deleted.files | xargs -n 1 rm -fr
+	)
+fi
 
 # Update all plugins
 for name_plugin in $(sudo -s cat $final_path/lib/plugins/*/plugin.info.txt | grep url | awk -F':' '{print $3}');
 do
-		# Get a official plugin for dokuwiki, not update a no-official
-		sudo wget -nv --quiet "https://github.com/splitbrain/dokuwiki-plugin-${name_plugin}/zipball/master" -O "${name_plugin}.zip" -o /dev/null || true
-		if [ -s "${name_plugin}.zip" ]; then
-			sudo unzip ${name_plugin}.zip
-			sudo cp -a splitbrain-dokuwiki-plugin-${name_plugin}*/. "${final_path}/lib/plugins/${name_plugin}/"
-		fi
+	# Get a official plugin for dokuwiki, not update a no-official
+	sudo wget -nv --quiet "https://github.com/splitbrain/dokuwiki-plugin-${name_plugin}/zipball/master" -O "${name_plugin}.zip" -o /dev/null || true
+	if [ -s "${name_plugin}.zip" ]; then
+		sudo unzip ${name_plugin}.zip
+		sudo cp -a splitbrain-dokuwiki-plugin-${name_plugin}*/. "${final_path}/lib/plugins/${name_plugin}/"
+	fi
 done
 
-# Files owned by www-data can just read
-sudo chown -R root: $final_path
+#=================================================
+
+# Verify if existing file needs to be upgraded by comparing it's size to new file from package
+# If different, do a backup of existing file and overwrite with new file
+# 
+# Safe here as this file is only used by Yunohost. Dokuwiki cannot modified it.
+ynh_backup_if_checksum_is_different "$final_path/conf/local.protected.php"
+
+# Recalculate and store the config file checksum into the app settings
+ynh_store_file_checksum "$final_path/conf/local.protected.php"
+
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
+
+# Files owned by dokuwiki can just read
+chown -R root: $final_path
 
 # except for conf, data, some data subfolders, and lib/plugin, where dokuwiki must have write permissions
-if [ -d "${final_path}/data/media" ]; then
-	sudo chown -R $app:root $final_path/{data/attic,data/cache,data/index,data/locks,data/media*,data/meta,data/pages,data/tmp}
+#TODO compare rights with install script !!!
```

Comment:
Is this still relevant? :wink:

Reply:
Well ... I have not idea and I am not fluent enough in "Github / git" to be able to go back in time to see why this test has been added. That's why I wanted to work on permissions to be able to iron all these tests

Reply:
Edit: I am a bit tired so be careful with what you will read, there will be mistakes :)

About the linux permissions, there seems to be multiple ways of handling this. I add a look at different packages to get ideas to define the "best" ways of managing dokuwiki permissions and there is multiple ways of doing this

In the past, there was changes in the package that I cannot define
https://github.com/YunoHost-Apps/dokuwiki_ynh/commit/95b5d8a1baeff004679da17d3ac0063c28759a0a#diff-ad962c6fd65b725b71aa46c8a715127fL51


**I see multiple solutions**

1. Set `$final_path` to `root:root` and allow only some folder to `dokuwiki` user where it needs to write (mainly `data` and `conf` folder)
```
chown -R root:root $final_path
find $final_path -type d -print0 | xargs -0 chmod 0755
find $final_path -type f -print0 | xargs -0 chmod 0644
```

**pros**
- Really secure as read only by default for "other"

**cons**
- Everybody can read inside dokuwiki folder
- Need to set permission finely in `conf` folder to not open breach : See https://www.dokuwiki.org/install:permissions

Only theses files should be writable, some are not created on a fresh install. 
>    conf/ directory, following files must be writable by the web process:
>        local.php (and local.php.bak) for the installer and for subsequent web configuration to work.
>        users.auth.php and acl.auth.php for the ACL web configuration and usermanager to work.
>        plugins.local.php (and plugins.local.php.bak) for the Extension Manager


2. Set `$final_path` to `root:dokuwiki` and allow only some folder to `dokuwiki` user where it needs to write (mainly `data` and `conf` folder)
```
chown -R root:dokuwiki $final_path
find $final_path -type d -print0 | xargs -0 chmod 0750
find $final_path -type f -print0 | xargs -0 chmod 0640
```

**pros**
- Secure as read only by default for group "dokuwiki"
- No other app can read dokuwiki data

**cons**
- Not tested so it might not work as "nginx" runs as "www-data" and as no read access on the folder
- Need to set permission finely in `conf` folder to not open breach : See https://www.dokuwiki.org/install:permissions


3. Set `$final_path` to `dokuwiki:dokuwiki` and allow only some folder to `dokuwiki` user where it needs to write (mainly `data` and `conf` folder)
```
chown -R dokuwiki:dokuwiki $final_path
find $final_path -type d -print0 | xargs -0 chmod 0750
find $final_path -type f -print0 | xargs -0 chmod 0640
```

**pros**
- No issue with permissions as "Dokuwiki" has write access

**cons**
- Might be less secure as "Dokuwiki" can write everywhere
- Need to set permission finely in `conf` folder to not open breach : See https://www.dokuwiki.org/install:permissions

4. Same as 3. but with read access to group other
```
chown -R dokuwiki:dokuwiki $final_path
find $final_path -type d -print0 | xargs -0 chmod 0755
find $final_path -type f -print0 | xargs -0 chmod 0645
```

Reply:
Without hesitation, I'd say first solution. It's better if root owns everything but what's really needed by the process' user.
The main security issue is if the process itself is corrupt, so that it should has write permission only where it's really needed.